balancer-js/src/modules/pools/pool-types/concerns/stable/join.concern.ts

```diff
@@ -55,32 +113,51 @@ export class StablePoolJoin implements JoinConcern {
       parsedSwapFee,
       scalingFactors,
       upScaledBalances,
-    } = parsePoolInfo(pool);
+    } = parsePoolInfo(pool, wrappedNativeAsset);
 
     const assetHelpers = new AssetHelpers(wrappedNativeAsset);
     // sort inputs
-    const [sortedTokens, sortedAmountsIn] = assetHelpers.sortTokens(
+    const [, sortedAmountsIn] = assetHelpers.sortTokens(
```

Comment:
A bit of doc here explaining token ordering would be useful. I think there's something in ComposableStable you could use.